src/main/java/pl/sda/weather/service/impl/WeatherServiceImpl.java

```diff
@@ -1,29 +1,38 @@
 package pl.sda.weather.service.impl;
 
-import pl.sda.weather.model.LocationModel;
-import pl.sda.weather.model.Weather;
+import pl.sda.weather.model.entity.LocationModelEntity;
+import pl.sda.weather.model.entity.WeatherModelEntity;
 import pl.sda.weather.repository.IWeatherRepository;
 import pl.sda.weather.repository.impl.WeatherRepositoryImpl;
 import pl.sda.weather.service.IWeatherService;
 
-import java.util.LinkedList;
 import java.util.List;
 
 public class WeatherServiceImpl implements IWeatherService {
 
-    private static IWeatherRepository readWeatherRepository = new WeatherRepositoryImpl();
+    IWeatherRepository readWeatherRepository = new WeatherRepositoryImpl();
 
     @Override
-    public List<Weather> listWeathers(List<LocationModel> citiesList) {
-        List<Weather> weatherList = new LinkedList<>();
-        for (LocationModel city : citiesList) {
-            weatherList.add(readWeatherRepository.readWeather(city.getCityName()));
+    public void listWeathers(List<LocationModelEntity> citiesList) {
+        cleanRecords();
+        for (LocationModelEntity city : citiesList) {
```

Comment:
in location service there is a public  method  List <LocationModelEntity> getAllLocation ()   you can use here instead of loops